dragon/docs.rb

```diff
@@ -267,6 +270,37 @@ def __docs_to_html__ string
 S
     html_content_end_to_html_end = <<-S
     </div>
+    <script src="prism.js" data-manual></script>
+    <script>
+      document.addEventListener("DOMContentLoaded", function() {
+              var lazyloadElements = document.querySelectorAll("code");
+              var lazyloadThrottleTimeout;
+
+              function lazyload () {
+                      if(lazyloadThrottleTimeout) {
+                              clearTimeout(lazyloadThrottleTimeout);
+                            }
+
+                      lazyloadThrottleTimeout = setTimeout(function() {
+                              var scrollTop = window.pageYOffset;
+                              lazyloadElements.forEach(function(elem) {
+                                      if(elem.offsetTop < (window.innerHeight + scrollTop)) {
+                                              Prism.highlightElement(elem);
+                                            }
+                                    });
+                              if(lazyloadElements.length == 0) {
+                                      document.removeEventListener("scroll", lazyload);
+                                      window.removeEventListener("resize", lazyload);
+                                      window.removeEventListener("orientationChange", lazyload);
+                                    }
+                            }, 20);
+                    }
+
+              document.addEventListener("scroll", lazyload);
+              window.addEventListener("resize", lazyload);
+              window.addEventListener("orientationChange", lazyload);
+            });
```

Comment:
At the moment this code is iterating over all code blocks every 20ms right?

I was thinking, how about maybe a version which just highlights all elements with a 100ms (or longer) timeout per element? That should be still performant enough for most cases.....

Only problem with this approach might be when re-opening a tab or a bookmark at a quite low position in the document, then it might take a few seconds until it reaches your screen position....

Reply:
So in the latest iteration I'm executing `highlightElement` on each `pre>code` element with a spaced delay after another delay from DOMContentLoaded.